pkg/products/monitoring/reconciler.go

```diff
@@ -776,3 +773,28 @@ func (r *Reconciler) reconcileSubscription(ctx context.Context, serverClient k8s
 	)
 
 }
+
+func (r *Reconciler) getDMSSecret(ctx context.Context, serverClient k8sclient.Client) (string, error) {
+
+	var secret string
+
+	dmsSecret := &corev1.Secret{}
+	err := serverClient.Get(ctx, types.NamespacedName{Name: r.installation.Spec.DeadMansSnitchSecret,
+		Namespace: r.installation.Namespace}, dmsSecret)
+
+	if err != nil {
+		return "", fmt.Errorf("could not obtain dead mans snitch credentials secret: %w", err)
+	}
+
+	if len(dmsSecret.Data["SNITCH_URL"]) != 0 {
+		secret = string(dmsSecret.Data["SNITCH_URL"])
+	} else if len(dmsSecret.Data["url"]) != 0 {
+		secret = string(dmsSecret.Data["url"])
+	}
```

Comment:
Would it work if we placed an `else` statement here and returned if the secret hasn't been assigned? i.e:

 ```
  else {
   return secret, fmt.Errorf("url is undefined in dead mans snitch secret")
  }
```

Reply:
It would work. Agreed and moved the code.